client/shared/src/test/scala/org/http4s/client/ClientSuite.scala

```diff
@@ -69,18 +67,6 @@ class ClientSpec extends Http4sSuite with Http4sDsl[IO] {
       .assertEquals("http4s.org:1983")
   }
 
-  test("mock client should cooperate with the VirtualHost server middleware") {
-    val routes = HttpRoutes.of[IO] { case r =>
-      Ok(r.headers.get[Host].map(_.value).getOrElse("None"))
-    }
-
-    val hostClient = Client.fromHttpApp(VirtualHost(exact(routes, "http4s.org")).orNotFound)
-
-    hostClient
-      .expect[String](Request[IO](GET, uri"https://http4s.org/"))
-      .assertEquals("http4s.org")
-  }
```

Comment:
This was moved to a JVM-only file because we are not crossing `server` for JS in 0.23.